### PR TITLE
Cleanup for code fixes

### PR DESCRIPTION
**Pull request by vweijsters**
Refactored all code fixes for consistency, based on discussions in #1441

### CODE REVIEW

**Comment by sharwell** [comment]
I find myself wishing this was multiple commits:

1. The first commit could simply wrap the calls to `RegisterCodeFix`
2. The second commit could remove the `.Where` and `if (...)` conditions that check to see if a diagnostic ID is fixable
3. The third commit could remove `FixableDiagnostics` as a separate field (favoring the `{ get; } = ...` syntax instead)

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1501UnitTests.cs [hunk 1]
```diff
@@ -507,12 +507,12 @@ void MethodName()
...
         do
         {
-            Bar();
+            MethodName();
         }
         while (false);
 ";
 
-            await this.VerifyCSharpFixAsync(testCode, fixedTestCode).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode, allowUnsupportedDiagnostics: true).ConfigureAwait(false);
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1501UnitTests.cs]: :question: Why did this one need to be a special case? It seems like a bunch of special behavior was added to the testing framework just to support this single test.

Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1501UnitTests.cs]: This test contains invalid C# syntax. The closing brace for the method is not present, resulting in a compiler being produced for it in the verify case.
As I wanted to keep the `Assert` in the `CodeFixVerifier` signalling that a test case is producing unexpected errors, I had to go to quite a bit of effort to make this test case work.

**Comment by vweijsters** [comment]
I'll address this in separate pull requests. Keeping this one open until I'm done with those

**Comment by vweijsters** [comment]
Closing this as all content has been addressed in other pull requests.

**Pull request closed without merging**